design/object-bucket/object-bucket.md

```diff
@@ -43,115 +51,159 @@ Add a CephObjectBucketClaim API endpoint and control loop to the Rook-Ceph opera
+- A generic S3 user API is not defined in this proposal but should be given serious consideration.  A user API should allow cluster users to generate new S3 users, tenants, and access keys in an object store.  This API could be used as method of manually creating object store users or incorporated into the bucket provisioning automation to generate the user's access key.
 
 ---
 
 ## API Specifications
 
-### Custom Resource Definition
+### OBC Custom Resource Definition
 
 ```yaml
 apiVersion: apiextensions.k8s.io/v1beta1 
 kind: CustomResourceDefinition
 metadata:
-  name: cephobjectbucketclaims.rook.io
+  name: objectbucketclaims.rook.io
 spec:
-  group: ceph.rook.io
+  group: rook.io
   version: v1
   scope: namespaced
   names:
-      kind: CephObjectBucketClaim
-      listKind: CephObjectBucketClaimList
-      singular: cephobjectbucketclaim
-      plural: cephobjectbucketclaims
+      kind: ObjectBucketClaim
+      listKind: ObjectBucketClaimList
+      singular: objectbucketclaim
+      plural: objectbucketclaims
 ```
 
 
-### Custom Resource (User Defined)
+### OBC Custom Resource (User Defined)
 
 ```yaml
-apiVersion: ceph.rook.io/v1
-kind: CephObjectBucketClaim
+apiVersion: rook.io/v1
+kind: ObjectBucketClaim
 metadata:
   name: MY-BUCKET-1 [1]
   namespace: USER-NAMESPACE [2]
 spec:
   storageClassName: SOME-OBJECT-STORE-STORAGE-CLASS [3]
+  tenant: MY-TENANT
 ```
 
-1. `name` of the CephObjectBucketClaim. Composes part of the bucket name and ConfigMap name
-1. `namespace` of the CephObjectBucketClaim. Composes part of the bucket name. Determines the namespace of the ConfigMap
+1. `name` of the ObjectBucketClaim. Composes part of the bucket name and ConfigMap name
+1. `namespace` of the ObjectBucketClaim. Composes part of the bucket name. Determines the namespace of the ConfigMap
 1. `storageClassName` is used to target the desired Object Store.  Used by the operator to get the Object Store service URL.
+1. `tenant` allows users to define a tenant in an object store in order to namespace their buckets and access keys. 
 
-### Custom Resource (Operator Updates, Post Creation)
+### OBC Custom Resource (Operator Updated)
 
 ```yaml
-apiVersion: ceph.rook.io/v1
-kind: CephObjectBucketClaim
+apiVersion: rook.io/v1
+kind: ObjectBucketClaim
 metadata:
   name: MY-BUCKET-1
   namespace: USER-NAMESPACE
-  ownerReferences: [1]
-  - name: SOME-CEPH-CLUSTER
-    kind: CephCluster
-    blockOwnerDeletion: true
-    <TRUCANTED FOR CLARITY>
-  labels:
-    ceph.rook.io/object/claims: [2]
 spec:
   storageClassName: SOME-OBJECT-STORE-STORAGE-CLASS
 status:
-  phase: {"pending", "available", "error"}  [3]
+  phase: {"pending", "bound", "error"}  [3]
   configMapRef: objectReference{}  [4]
+  secretRef: objectReference{}  [5]
```

Comment:
No labels, no annotations, no ownerReference added?

Reply:
Add `bucketReference` too?

Reply:
> No labels, no annotations, no ownerReference added?

This is something we need to sit down and hammer out.